### PR TITLE
feat: rendering chat_template

### PR DESCRIPTION
**Pull request by namchuai**
## Describe Your Changes

-

## Fixes Issues

- Closes #
- Closes #

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

File: engine/services/model_service.cc [hunk 1]
```diff
@@ -1208,3 +1223,39 @@ ModelService::MayFallbackToCpu(const std::string& model_path, int ngl,
...
+      file_manager_utils::ToAbsoluteCortexDataPath(
+          std::filesystem::path(file)));
+  if (!model_metadata_res.has_value()) {
+    CTL_ERR("Failed to read metadata: " + model_metadata_res.error());
+    return cpp::fail("Failed to read metadata: " + model_metadata_res.error());
+  }
+  return std::move(*model_metadata_res);
+}
+
+std::optional<std::shared_ptr<ModelMetadata>>
```
Comment (vansangpfiev) [inline engine/services/model_service.cc]: I think can use std::shared_ptr and return nullptr here

File: engine/utils/jinja_utils.h [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+#include <json/value.h>
+#include <string>
+
+#include "extensions/remote-engine/template_renderer.h"
+#include "utils/chat-template.hpp"
+#include "utils/result.hpp"
+
+namespace jinja {
```
Comment (vansangpfiev) [inline engine/utils/jinja_utils.h:10]: I think we use `inja` not `jinja`

**Review by vansangpfiev — APPROVED** [review]

File: engine/utils/jinja_utils.h (hunk 2 above)
Comment (namchuai) [reply engine/utils/jinja_utils.h:10]: I think jinja is more accurate in semantic. inja is a library tool.

File: engine/services/model_service.cc (hunk 1 above)
Comment (namchuai) [reply engine/services/model_service.cc]: ah yeah, thank you, let me update it.

**Pull request merged**